controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1358,6 +1360,110 @@ func (r *BareMetalHostReconciler) actionDeprovisioning(prov provisioner.Provisio
 	return actionComplete{}
 }
 
+func (r *BareMetalHostReconciler) checkServicing(prov provisioner.Provisioner, info *reconcileInfo, hup *metal3api.HostUpdatePolicy) (result actionResult, isServicing bool) {
+	servicingData := provisioner.ServicingData{}
+	// start janders mods
+	var fwDirty bool
+	var hfsDirty bool
+
+	// (NOTE)janders: since Servicing is an opt-in feature that requires HostUpdatePolicy to be created and set to onReboot
+	// set below booleans to false by default and change to true based on policy settings
+
+	servicingData.LiveFirmwareUpdateAllowed = false
+	servicingData.LiveFirmwareSettingsAllowed = false
+
+	info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). Defaults: LiveFirmwareUpdateAllowed: %+v LiveFirmwareSettingsAllowed:  %+v. (end of message)", servicingData.LiveFirmwareUpdateAllowed, servicingData.LiveFirmwareSettingsAllowed))
+	if hup == nil {
+		info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). HostUpdatePolicy is nil"))
+	}
+	if hup != nil {
+		info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). HostUpdatePolicy value:  %+v", hup))
+		if hup.Spec.FirmwareSettings == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareSettingsAllowed = true
+			info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). hup.Spec.FirmwareSettings set to OnReboot, setting LiveFirmwareSettingsAllowed to  %+v", servicingData.LiveFirmwareSettingsAllowed))
+		}
+		if hup.Spec.FirmwareUpdates == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareUpdateAllowed = true
+			info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). hup.Spec.FirmwareUpdates set to OnReboot, setting LiveFirmwareUpdateAllowed to  %+v", servicingData.LiveFirmwareUpdateAllowed))
+		}
+	}
+
+	if servicingData.LiveFirmwareUpdateAllowed {
```

Comment:
If I understood correctly `LiveFirmwareUpdateAllowed` will be related to the HFC, so you need to update things here to make it work, I've sent you a DM on Slack.